### PR TITLE
Implemented ComputeEpsilonSubset and added config to disable cudaDeviceSynchronize in debug

### PR DESCRIPTION
**Pull request by qindazhu**

### CODE REVIEW

File: k2/csrc/log.h [hunk 1]
```diff
@@ -214,13 +215,19 @@ class Voidifier {
...
+// The parameter of `K2_CUDA_SAFE_CALL` should be cuda function call or kernel
+// launch.
+// Noted we would never call `cudaDeviceSynchronize` in release mode and
+// user can even disable this call for debug mode by setting an environment
+// variable `K2_SYNC_KERNELS` with any non-empty value.
+#define K2_CUDA_SAFE_CALL(...)                     \
+  do {                                             \
+    __VA_ARGS__;                                   \
+    if (!::k2::internal::kDisableDebug &&          \
+        std::getenv("K2_SYNC_KERNELS") != nullptr) \
```
Comment (csukuangfj) [inline k2/csrc/log.h:227]: We can cache the result of `std::getenv` instead of invoking it for every call.

Comment (qindazhu) [reply k2/csrc/log.h:227]: OK, will fix tomorrow as I have to go home now..

**Comment by qindazhu** [comment]
Changed to WIP as I just realized I can get row splits with array index op, will update it tomorrow

File: k2/csrc/log.h (hunk 1 above)
Comment (qindazhu) [reply k2/csrc/log.h]: Fixed

**Comment by qindazhu** [comment]
Ready to review.

**Comment by danpovey** [comment]
Great!!  Merging.

**Pull request merged**